### PR TITLE
[Feature] Migrate csra to 1.x

### PR DESCRIPTION
**Pull request by yingfhu**
## Motivation

Migrate CSRA to 1.x.

## Modification

Add metrics `VOCMultiLabelMetric` and `VOCAveragePrecision` for VOC dataset because of the difficult attribute in objects.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: configs/csra/resnet101-csra_1xb16_voc07-448px.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    neck=None,
+    head=dict(
+        type='CSRAClsHead',
+        num_classes=20,
+        in_channels=2048,
+        num_heads=1,
+        lam=0.1,
+        loss=dict(type='CrossEntropyLoss', use_sigmoid=True, loss_weight=1.0)))
+
+# dataset setting
```
Comment (Ezra-Yu) [inline configs/csra/resnet101-csra_1xb16_voc07-448px.py:31]: please move all the dataset setting into 'config/_base_/datasets/voc_bs16.py'.

**Review by Ezra-Yu — CHANGES_REQUESTED** [review]

File: mmcls/evaluation/metrics/voc_multi_label.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            result['pred_score'] = pred_label['score'].clone()
+            num_classes = result['pred_score'].size()[-1]
+
+            if 'score' in gt_label:
+                result['gt_score'] = gt_label['score'].clone()
+            else:
+                result['gt_score'] = LabelData.label_to_onehot(
+                    gt_label['label'], num_classes)
+
+            difficult_label = set(gt_label_difficult) - (
```
[the remark this answers is not in the document]
Comment (yingfhu) [reply mmcls/evaluation/metrics/voc_multi_label.py:55]: True, added

File: configs/csra/resnet101-csra_1xb16_voc07-448px.py (hunk 1 above)
Comment (yingfhu) [reply configs/csra/resnet101-csra_1xb16_voc07-448px.py:31]: vgg also use this base config, cannot replace it

File: mmcls/evaluation/metrics/multi_label.py [hunk 3]
```diff
@@ -400,6 +400,12 @@ def _average_precision(pred: torch.Tensor,
     # a small value for division by zero errors
     eps = torch.finfo(torch.float32).eps
 
+    # get rid of -1 target such as difficult sample
```
[the remark this answers is not in the document]
Comment (yingfhu) [reply mmcls/evaluation/metrics/multi_label.py:403]: to ignore -1 index

File: mmcls/evaluation/metrics/voc_multi_label.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+from mmcls.registry import METRICS
+from .multi_label import AveragePrecision, MultiLabelMetric
+
+
+class VOCMetricMixin:
+    """A mixin class for VOC dataset metrics, VOC annotations have extra
+    `difficult` attribute for each object, therefore, extra option is needed
+    for calculating VOC metrics.
+
```
[the remark this answers is not in the document]
Comment (yingfhu) [reply mmcls/evaluation/metrics/voc_multi_label.py:14]: there are three options for difficult label, -1 for ignorance, 0 for negative, 1 for positive

**Review by mzr1996 — APPROVED** [review]

**Pull request merged**